### PR TITLE
33944- allows txt and png files to be uploaded

### PR DESCRIPTION
**Pull request by alsleaders**
## Description
While testing file uploads for supporting evidence for Form 526, .png and .txt files were failing, even though we say we support uploading them 

## Original issue(s)
[department-of-veterans-affairs/va.gov-team#33944](https://github.com/department-of-veterans-affairs/va.gov-team/issues/33944)

## Testing done
Added unit test

## Acceptance criteria
- [ ] All tests passing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by Mottie — APPROVED** [review]
Looks good! Thanks for doing this work!

**Review by Mottie — APPROVED** [review]

File: src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js [hunk 1]
```diff
@@ -89,13 +89,16 @@ export default function checkTypeAndExtensionMatches({ file, result }) {
...
+  if (typeof signature !== 'undefined' && match?.mime === file.type) {
+    if (extension === 'txt') return true;
+    else {
+      return arrayIncludesArray(
+        result,
+        Array.isArray(signature)
+          ? signature
+          : [...signature].map(str => str.charCodeAt(0)),
+      );
+    }
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js:101]: What is `typeof signature !== 'undefined'` doing for us that the truthy check wasn't covering?

Also, could we get a unit test for this added in here?

https://github.com/department-of-veterans-affairs/vets-website/blob/63d0930f8e22a41288d8176edaf3e6353f280883/src/platform/forms-system/test/js/utilities/file.unit.spec.js#L78

Comment (alsleaders) [reply src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js]: That check wasn't doing what I thought it was (fixing the problem uploading .png files) so I've removed it and reworked the `extension` check slightly.  Also I've added a unit test to that file.

File: src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js [hunk 2]
```diff
@@ -89,13 +89,13 @@ export default function checkTypeAndExtensionMatches({ file, result }) {
   const match = extension && fileTypeSignatures[extension];
   const signature = match?.sig;
 
-  if (signature && match?.mime === file.type) {
+  if (extension === 'txt') return true;
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js:92]: Should this also check that `file.type` is txt or something along those lines? What if someone were to append `.txt` onto the end of a binary file?

Comment (Mottie) [reply src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js]: Please take a look when you get a chance. Thanks!

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**